src/utils/generator.js

```diff
@@ -1253,3 +1325,108 @@ export class PreludeGenerator {
     return ref;
   }
 }
+
+type TemporalBuildNodeEntryOptimizationStatus = "NO_OPTIMIZATION" | "POSSIBLE_OPTIMIZATION";
+
+// This function attempts to optimize Object.assign calls, by merging mulitple
+// calls into one another where possible. For example:
+//
+// var a = Object.assign({}, someAbstact);
+// var b = Object.assign({}, a);
+//
+// Becomes:
+// var b = Object.assign({}, someAbstract, a);
+//
+export function attemptToMergeEquivalentObjectAssigns(
+  realm: Realm,
+  callbacks: VisitEntryCallbacks,
+  temporalBuildNodeEntry: TemporalBuildNodeEntry
+): TemporalBuildNodeEntryOptimizationStatus | TemporalObjectAssignEntry {
+  let args = temporalBuildNodeEntry.args;
+  // If we are Object.assigning 3 or more args
+  if (args.length < 3) {
+    return "NO_OPTIMIZATION";
+  }
+  let objectAssingFunc = args[0];
+  let to = args[1];
+  // Then scan through the args after the "to" of this Object.assign, to see if any
+  // other sources are the "to" of a previous Object.assign call
+  loopThroughArgs: for (let i = 2; i < args.length; i++) {
+    let possibleOtherObjectAssignTo = args[i];
+    // Ensure that the "to" value can be omitted
+    if (!callbacks.canOmit(possibleOtherObjectAssignTo)) {
```

Comment:
I maintain that this check if fragile and depends on the visiting order, but concede that it's not a functional problem. Leave behind a comment to this end.